src/home-page/Header/Header.tsx

```diff
@@ -1,25 +1,41 @@
-import { useState } from 'react';
+import { useState, useContext, useCallback } from 'react';
 
-import { Modal, AddMovieModal } from '@shared';
+import { Modal, AddMovieModal, MovieDetailsContext } from '@shared';
 
 import { HeaderTopWrapper } from './HeaderTop';
 import { SearchWrapper } from './HeaderSearch';
-import { HeaderWrapper } from './HeaderWrapper';
+import { HeaderWrapper } from './Header.styles';
+import { MovieDetails } from './MovieDetails';
 
 export const HomePageHeader = () => {
   const [isOpen, setIsOpen] = useState(false);
+  const { movie, showMovieDetails } = useContext(MovieDetailsContext);
+
+  const hideMovieDetails = useCallback(() => showMovieDetails(null), []);
```

Comment:
I may be missing something, but these event handlers don't seem to depend on anything _external_, so they will trigger state update and consequent re-render, despite of referential equality. Considering `useMemo()` / `useCallback()` optimization [comes at a cost](https://kentcdodds.com/blog/usememo-and-usecallback), what's the point using them here?

Reply:
That's a top component so I declare callbacks here, so children don't get re-rendered. Anyway, it was a part of HW, I wouldn't use this hook here anyway.